lib/active_merchant/billing/gateways/moneris_us.rb

```diff
@@ -154,14 +157,39 @@ def expdate(creditcard)
         sprintf("%.4i", creditcard.year)[-2..-1] + sprintf("%.2i", creditcard.month)
       end
 
+      def add_address(post, payment_method, options)
+        if !payment_method.is_a?(String) && card_brand(payment_method) == 'check'
+          post[:ach_info][:cust_first_name] = payment_method.first_name if payment_method.first_name
```

Comment:
Does it make sense to `post[:ach_info] ||= {}` here? If someone reordered `add_address` and `add_payment_source` calls in `purchase` or something, it'd suddenly crash, and the lack of unit tests means that might not be caught super quickly. (Alternatively, if there were a purchase unit test, I think I'd be okay with this arrangement.)

Reply:
@bpollack Went ahead and added the unit test.